### ISSUE TITLE
[WIIU] Regression: Keyboard input broken, all unrecognized USB devices recognized as Generic SNES USB controllers.

### ISSUE THREAD

**Issue by vaguerant** [libretro/RetroArch#13198]
## Description

This was already flagged to @gblues over in the GBATemp thread for the Wii U version, but I've finally done some further testing on it, so here's a proper issue report.

Since the USB HID rewrite on Wii U (started with PR #13073), any unrecognized USB HID device is detected as a "Generic SNES USB controller" and does not work. This has also broken keyboard input, which used to work without RetroArch needing to recognize the specific keyboard in question.

Connecting more than one unrecognized USB device also causes the port 1 input to disconnect, meaning the non-existent SNES controllers are the only connected controllers. Leaving the unrecognized device/s unplugged prevents the secondary issues from occurring (the port 1 disconnect/phantom SNES pads), but obviously doesn't help with getting USB keyboards working again in RetroArch.

### Expected behavior

Unrecognized USB HID devices (including keyboards, which used to work) shouldn't be mapped as SNES controllers or anything else.

### Actual behavior

Unrecognized USB HID devices are mapped as SNES controllers which are "not configured" and if more than one are attached, the port 1 controller (generally the Wii U GamePad) is disconnected as a controller--the GamePad is still connected in the sense that it is powered on and mirroring the TV, it just isn't a connected controller in RetroArch any more. Attaching an additional compatible controller (e.g. Wii Remote) does assign it to port 1, allowing for recovery from this situation, e.g. by exiting RetroArch, unplugging the unrecognized device, then returning.

The port 1 disconnection can be triggered either by physically inserting multiple unrecognized devices, or by connecting any USB device which exposes multiple unrecognized sub-devices. e.g. Testing with a single USB controller or USB mouse did not cause a port 1 disconnect, but connecting both at the same time did. Similarly, I tested a wireless combined mouse and keyboard which share a single adapter: with this device and nothing else connected, I get two "Generic SNES USB controller" connections and the port 1 controller disconnects. @Ploggy in the [original GBATemp thread](https://gbatemp.net/threads/retroarch-wiiu-wip.447670/post-9631783) noted the same port 1 disconnection with their Mayflash USB adapter. I'm not sure what type of adapter Ploggy owns, but most of Mayflash's products are multi-port adapters, i.e. they are similarly exposing multiple devices.

### Additional notes

I'd argue there's two roughly three bugs here: 

1) the RetroArch Wii U HID implementation is being applied to all HID devices, not just HID gamepads/joysticks which seems to be the actual goal. HID keyboards shouldn't really need any special handling, they're fundamentally all the same, and the intention of anyone plugging one into a Wii U is not to use it as a gamepad--they're more useful for the various home computer emulator cores, e.g. Commodore 64 emulators. HID mice are not currently supported in RetroArch Wii U, but I'd say they fall into the same family of "devices nobody is trying to use as gamepads," so they should also be exempted from the USB HID game controller handling.
2) unrecognized HID devices are somehow being misidentified as Generic SNES USB controllers.
3) connecting more than one unrecognized HID device will disconnect the (real) controller that is connected in port 1.

Attached is a log from when I first flagged this issue a couple of weeks ago, but I have confirmed that behavior remains the same on current builds.

[retroarch.zip](https://github.com/libretro/RetroArch/files/7492104/retroarch.zip)

### Steps to reproduce the bug

1. Have one or more unrecognized USB HID devices connected to the Wii U.
2. Launch RetroArch.

or
1. Launch RetroArch.
2. Connect one or more unrecognized USB HID devices connected to the Wii U.

### Bisect Results

f242ef6 originally broke things worse than this, having an unrecognized device attached would actively crash RetroArch immediately on launch. This was improved by a later PR (probably one of dd6c70a or 2f4cfe2?) which prevented the crash but either caused or exposed the issue here. Since the core wouldn't run at all with unrecognized devices attached in the intervening period, it's hard to say when *exactly* this issue first started.

### Version/Commit

- RetroArch: a93a2e3

### Environment information

- OS: Wii U Menu 5.5.5 PAL
- Compiler: official buildbot libretro-build-wiiu Docker image

**Comment by gblues** [comment]
so PR #13308 fixes the keyboard side of things, and should fix this for certain classes of unrecognized USB HID devices--mice and keyboards.

Once that PR is merged, please re-test, and if the issue persists, please file a separate issue and collect some logs using the `wiiu/net_listen.sh` script (python required). I reviewed the code and I'm not sure why it's landing on the "Generic SNES" device and I'm hoping the logs will shed some light.

**Comment by vaguerant** [comment]
Improved by #13308, continued in #13309.
